src/main/java/com/google/devtools/build/lib/skyframe/IgnoredPackagePrefixesFunction.java

```diff
@@ -40,12 +42,23 @@
 /**
  * A {@link SkyFunction} for {@link IgnoredPackagePrefixesValue}.
  *
- * <p>It is used to implement the `.bazelignore` feature.
+ * <p>It is used to compute which directories should be ignored in a package. These either come
+ * from the {@code .bazelignore} file or from the {@code ignored_directories()} function in
+ * {@code REPO.bazel}.
  */
 public class IgnoredPackagePrefixesFunction implements SkyFunction {
+  /**
+   * A version of {@link IgnoredPackagePrefixesFunction} that always returns the empty value.
+   *
+   * <p> Used for tests where the extra complications incurred by evaluating the function are
+   * undesired.
+   */
+  public static final SkyFunction NOOP = (skyKey, env) -> IgnoredPackagePrefixesValue.EMPTY;
+
   private final PathFragment ignoredPackagePrefixesFile;
 
-  public IgnoredPackagePrefixesFunction(PathFragment ignoredPackagePrefixesFile) {
+  public IgnoredPackagePrefixesFunction(
+      PathFragment ignoredPackagePrefixesFile) {
```

Comment:
sorry to keep harping on about this -- but just in case I didn't get the point through, I don't think this needs to be a ctor argument at all. Its value is only ever `PathFragment.create(".bazelignore")`. Instead of having it passed in by BazelSkyframeExecutorConstants and stored as a member, we can just move the constant into this class and refer to it at lines 132 and 152.

Essentially, I'm approaching this code as if I'm writing it the first time. With that mindset, it's very weird to me that this constant is passed in as a ctor argument. (Yeah I know that it was becuase `NOOP` was implemented using this, but it's no longer done like that.) So I'd like to avoid surprising future readers.

Reply:
If the argument was always `.bazelignore`, you were totally right. Sadly, that's not the case at least in three places: `PrepareDepsOfPatternsFunctionSmartNegationTest.java:100`, `SkyframeQueryHelper.createSkyframeExecutor.java:428` and `PackageLookupFunctionTest.java:143`. 

So I'd much rather not rock the boat. Let me see if I have time today to track these down after importing this change...

Reply:
ah, I see. Those are for testing purposes, so technically still not needed, but I'm happy to leave those for later.